docs/development/languages/how-to-write-go-code.md

```diff
@@ -38,3 +38,54 @@ func DoSomething() (string, error) {
 	return "", fmt.Errorf("%w: %s", ErrUpstreamFailure, fmt.Errorf("upstream error"))
 }
 ```
+
+## Object Construction
+
+When creating constructors for objects (structs) in Go, prefer the **Functional Options Pattern**. This approach uses variadic arguments to handle optional configuration while keeping the API clean and extensible.
+
+*   **Required Arguments:** Pass required dependencies as explicit, typed arguments to the constructor function (e.g., `NewServer`).
```

Comment:
If a package primarily creates a single object (for example a server), just naming the construction function New is fine. Server.newserver is a stutter.